### PR TITLE
Fixed no payment/shipping method fixtures

### LINKED ISSUE #10490: Make it possible to have no shipping methods in fixtures
**Sylius version affected**: 1.4

**Description**  
You can't use Order fixtures if you don't have any shipping method fixtures.

**Steps to reproduce**  

Create a fixture suite with orders but without shipping methods.

**Possible Solution**  

#10489

### LINKED ISSUE #10502: Make it possible to have no payment methods in fixtures
**Sylius version affected**: 1.6/master (but I guess any of 1.*)

**Description**  
Similar issue as #10490, but for Payment methods.

### PR DESCRIPTION
**Pull request by igormukhingmailcom**
| Q               | A
| --------------- | -----
| Branch?         | master (sorry, should be based to 1.4)
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #10502, fixes #10490 right way
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Fixture/OrderFixture.php [hunk 1]
```diff
@@ -216,34 +217,53 @@ private function address(OrderInterface $order, string $countryCode): void
...
+            $shippingMethod = $this
+                ->faker
+                ->randomElement($this->shippingMethodRepository->findEnabledForChannel($order->getChannel()))
+            ;
+
+            /** @var ChannelInterface $channel */
+            $channel = $order->getChannel();
+            Assert::notNull($shippingMethod, sprintf(
+                "No enabled shipping method was found for channel '%s'. " .
+                "Set 'skipping_shipping_step_allowed' option to true for this channel if you want to skip shipping method selection.",
```
Comment (TiMESPLiNTER) [inline src/Sylius/Bundle/CoreBundle/Fixture/OrderFixture.php]: 👍🏻

File: src/Sylius/Bundle/CoreBundle/Fixture/OrderFixture.php [hunk 2]
```diff
@@ -216,34 +217,53 @@ private function address(OrderInterface $order, string $countryCode): void
...
+                $channel->getCode()
+            ));
+
+            foreach ($order->getShipments() as $shipment) {
+                $shipment->setMethod($shippingMethod);
+            }
+
             $this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_SELECT_SHIPPING);
+        } else {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Fixture/OrderFixture.php]: We could just use early return and skip `else` statement. Also, maybe reverse the `if` condition and put `$this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_SKIP_SHIPPING);` inside, which would make it easier to understand (IMHO). 

WDYT?

**Comment by lchrusciel** [comment]
Also, PR need rebasing. Once we will fix it, I'll change base branch. Now it could do more problems to fix current problems

**Comment by Zales0123** [comment]
@igormukhingmailcom can you take a look at this? also it would be great to squash some commits

**Comment by Zales0123** [comment]
Replaced by https://github.com/Sylius/Sylius/pull/10618

**Pull request closed without merging**